src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -75,22 +73,9 @@ public void handle(HttpExchange exchange) throws IOException {
                                         req.signature);
 
                         if (Objects.nonNull(response)) {
-                            // If a thread dump was requested we need to send it back
-                            if (req.operation.equals(DUMP_THREADS)) {
-                                exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_UNKNOWN);
-                                try (OutputStreamWriter writer =
-                                        new OutputStreamWriter(
-                                                exchange.getResponseBody(), Charsets.UTF_8)) {
-                                    writer.write(response.toString());
-                                } catch (Exception e) {
-                                    log.error("Failed to write thread dump to response: ", e);
-                                    throw e;
-                                }
-                            } else {
-                                exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_UNKNOWN);
-                                try (OutputStream responseStream = exchange.getResponseBody()) {
-                                    mapper.writeValue(responseStream, response);
-                                }
+                            exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_UNKNOWN);
+                            try (OutputStream responseStream = exchange.getResponseBody()) {
+                                mapper.writeValue(responseStream, response);
```

Comment:
I'm not sure if this is what causes the problem since I don't remember if I had tested earlier revisions of this work with the Agent in particular, but now that I am testing I find that the thread dumps retrieved via JMX are formatted how I expect them to be, but the ones retrieved via Agent HTTP don't look right.